### PR TITLE
Update how func-test-pr ends up being calculated

### PR DESCRIPTION
**Pull request by ChrisMacNaughton**

### CODE REVIEW

**Review by AurelienLourot — APPROVED** [review]
This looks as correct as before to me :D What's the difference?

**Comment by ChrisMacNaughton** [comment]
What I'm trying here is to only read the curl output once, and setting it's output as a fact if it's present. I'm hoping this can resolve the issue that currently hits func-test-pr usage where we get logs that look like:

```
2021-03-01 12:28:27.264024 | PLAY [all]
2021-03-01 12:28:27.302401 | 
2021-03-01 12:28:27.302788 | TASK [has func-test-pr]
2021-03-01 12:28:27.853404 | bionic-medium | openstack-charmers/zaza-openstack-tests/369
2021-03-01 12:28:27.896738 | bionic-medium | ok: Runtime: 0:00:00.009143
2021-03-01 12:28:27.936376 | 
2021-03-01 12:28:27.936748 | TASK [func-test repo]
2021-03-01 12:28:28.289967 | bionic-medium | openstack-charmers/zaza-openstack-tests
2021-03-01 12:28:28.487911 | bionic-medium | ok: Runtime: 0:00:00.003585
2021-03-01 12:28:28.515900 | 
2021-03-01 12:28:28.516117 | TASK [Github PR repo]
2021-03-01 12:28:28.855925 | bionic-medium | zaza-openstack-tests
2021-03-01 12:28:29.063383 | bionic-medium | ok: Runtime: 0:00:00.003836
2021-03-01 12:28:29.090734 | 
2021-03-01 12:28:29.090964 | TASK [Github PR org]
2021-03-01 12:28:29.466958 | bionic-medium | openstack-charmers
2021-03-01 12:28:29.637061 | bionic-medium | ok: Runtime: 0:00:00.004206
2021-03-01 12:28:29.666130 | 
2021-03-01 12:28:29.666423 | TASK [func-test PR number]
2021-03-01 12:28:30.008520 | bionic-medium | 369
2021-03-01 12:28:30.214601 | bionic-medium | ok: Runtime: 0:00:00.003867
2021-03-01 12:28:30.250859 | 
2021-03-01 12:28:30.251149 | TASK [func-test pr label]
2021-03-01 12:28:30.604546 | bionic-medium |   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
2021-03-01 12:28:30.604608 | bionic-medium |                                  Dload  Upload   Total   Spent    Left  Speed
2021-03-01 12:28:31.355040 | bionic-medium | 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 19346  100 19346    0     0  25794      0 --:--:-- --:--:-- --:--:-- 25794
2021-03-01 12:28:31.357668 | bionic-medium | auria:hacluster-scaleback
2021-03-01 12:28:31.802486 | bionic-medium | ok: Runtime: 0:00:00.765701
2021-03-01 12:28:31.831145 | 
2021-03-01 12:28:31.831377 | TASK [Github PR login]
2021-03-01 12:28:32.171853 | bionic-medium | % Total % Received % Xferd Average Speed Time Time Time Current
2021-03-01 12:28:32.171899 | bionic-medium | /bin/sh: 2: Dload: not found
2021-03-01 12:28:32.171909 | bionic-medium | /bin/sh: 3: 
: not found
2021-03-01 12:28:32.172149 | bionic-medium | /bin/sh: 4: auria:hacluster-scaleback: not found
2021-03-01 12:28:32.376667 | bionic-medium | ok: Runtime: 0:00:00.004008
2021-03-01 12:28:32.404861 | 
2021-03-01 12:28:32.405076 | TASK [Github PR branch]
2021-03-01 12:28:32.724847 | bionic-medium | % Total % Received % Xferd Average Speed Time Time Time Current
2021-03-01 12:28:32.724906 | bionic-medium | /bin/sh: 2: Dload: not found
2021-03-01 12:28:32.724916 | bionic-medium | /bin/sh: 3: 
: not found
2021-03-01 12:28:32.726895 | bionic-medium | /bin/sh: 4: auria:hacluster-scaleback: not found
2021-03-01 12:28:32.950263 | bionic-medium | ok: Runtime: 0:00:00.005229
```

File: playbooks/juju/pre.yaml [hunk 1]
```diff
@@ -36,21 +36,25 @@
...
       shell: >
-        echo {{ func_pr_label.stdout }} | cut -f2 -d: | sed -e 's/\//\\\//g'
+        echo {{ func_pr_label }} | cut -f2 -d: | sed -e 's/\//\\\//g'
       register: github_pr_branch
 
     - name: 'handle func-test-pr'
-      when: github_pr_login.rc
+      when: github_pr_login.rc == 0
       shell: >
         GITHUB_PR_REPO="{{ github_pr_repo.stdout }}"
```
Comment (AurelienLourot) [inline playbooks/juju/pre.yaml:59]: I think the issue is that I don't see where `github_pr_repo` is being defined and the particular func-test-pr we're looking at is coming from a fork (so a different repo). What do you think?

**Review by AurelienLourot — CHANGES_REQUESTED** [review]

File: playbooks/juju/pre.yaml (hunk 1 above)
Comment (AurelienLourot) [reply playbooks/juju/pre.yaml:59]: Ah I see, line 19, but maybe you want to do the same trick there?

Comment (ChrisMacNaughton) [reply playbooks/juju/pre.yaml:59]: We're only reading the stdout from `github_pr_repo` once, where before we were actually using the `func-pr-label` twice. I don't know that setting it two different times will really buy us much here

**Review by AurelienLourot — APPROVED** [review]

**Pull request merged**